### ISSUE TITLE
Tehran wrong TZ info (+0430) for summer DST

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#21991]
export TZ=Asia/Tehran

```
$ date
Mon 22 Mar 2021 08:46:47 PM +0430

SELECT toTimeZone(now(), 'Asia/Tehran')
┌─toTimeZone(now(), 'Asia/Tehran')─┐
│              2021-03-22 19:47:04 │
└──────────────────────────────────┘
```

related https://github.com/ClickHouse/ClickHouse/issues/13845

**Comment by alexey-milovidov** [comment]
The version 21.3 is also alright.

**Comment by alexey-milovidov** [comment]
21.2 is good as well.

**Comment by alexey-milovidov** [comment]
Reproduced in 21.3:
```
milovidov@milovidov-desktop:~/work/ClickHouse$ programs/server/clickhouse-21.3.3 local --query "SELECT toTimeZone(now(), 'Asia/Tehran')"
2021-03-22 22:11:42
```

And it is perfectly alright in master:
```
milovidov@milovidov-desktop:~/work/ClickHouse$ build/programs/clickhouse local --query "SELECT toTimeZone(now(), 'Asia/Tehran')"
2021-03-22 23:11:44
```

It has been fixed in #9404
The change will be shipped in version 21.4 in April.

The fix is very complicated and not going to be backported to any of the previous releases.

**Comment by alexey-milovidov** [comment]
I will add a test just in case...
